### PR TITLE
port BLS primitives to using BLST

### PR DESCRIPTION
**Pull request by arvidn**
testing this on Apple M1 yields a material speedup:
```
     Running benches/derive_key.rs (/Users/arvid/Documents/dev/chia_rs/target/release/deps/derive_key-319a95fb53fa0c95)
secret key, unhardened  time:   [87.483 µs 87.737 µs 88.046 µs]
                        change: [-71.923% -71.826% -71.742%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 1 outliers among 100 measurements (1.00%)
  1 (1.00%) high mild

secret key, hardened    time:   [440.89 µs 441.48 µs 442.14 µs]
                        change: [+1.3383% +1.9851% +2.7497%] (p = 0.00 < 0.05)
                        Performance has regressed.
Found 4 outliers among 100 measurements (4.00%)
  1 (1.00%) high mild
  3 (3.00%) high severe

public key, unhardened  time:   [92.672 µs 92.827 µs 93.004 µs]
                        change: [-70.570% -70.500% -70.433%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 9 outliers among 100 measurements (9.00%)
  1 (1.00%) low mild
  3 (3.00%) high mild
  5 (5.00%) high severe

     Running benches/sign.rs (/Users/arvid/Documents/dev/chia_rs/target/release/deps/sign-c9c2e6b916894092)
sign, small msg         time:   [389.90 µs 390.18 µs 390.48 µs]
                        change: [-81.308% -81.285% -81.259%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 4 outliers among 100 measurements (4.00%)
  2 (2.00%) high mild
  2 (2.00%) high severe

sign, 4kiB msg          time:   [391.07 µs 391.59 µs 392.15 µs]
                        change: [-81.351% -81.325% -81.297%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 2 outliers among 100 measurements (2.00%)
  1 (1.00%) high mild
  1 (1.00%) high severe

     Running benches/verify.rs (/Users/arvid/Documents/dev/chia_rs/target/release/deps/verify-0b14704012d8afbf)
verify, small msg       time:   [947.90 µs 948.84 µs 949.80 µs]
                        change: [-59.673% -59.624% -59.570%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 1 outliers among 100 measurements (1.00%)
  1 (1.00%) high severe

verify, 4kiB msg        time:   [948.81 µs 949.77 µs 950.79 µs]
                        change: [-59.875% -59.826% -59.775%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 3 outliers among 100 measurements (3.00%)
  2 (2.00%) high mild
  1 (1.00%) high severe
```

### CODE REVIEW

**Review by Rigidity — APPROVED** [review]
Looks good to me. I want to improve the error handling, but this is unrelated to changes in this PR, so I'll leave it for now.

**Review by Rigidity — APPROVED** [review]

File: chia-bls/src/signature.rs [hunk 1]
```diff
@@ -1,44 +1,67 @@
...
+            let mut bytes = MaybeUninit::<[u8; 96]>::uninit();
+            blst_p2_compress(bytes.as_mut_ptr() as *mut u8, &self.0);
+            bytes.assume_init()
+        }
     }
 
     pub fn aggregate(&mut self, sig: &Signature) {
-        self.0 += sig.0;
+        unsafe {
+            blst_p2_add(&mut self.0, &self.0, &sig.0);
```
Comment (emlowe) [inline chia-bls/src/signature.rs:48]: in bls-signature we used `blst_p2_add_or_double`

Comment (arvidn) [reply chia-bls/src/signature.rs:48]: oh, I've fixed this locally, but in a later commit, that I didn't include here. I'll fix it.

Comment (arvidn) [reply chia-bls/src/signature.rs]: I added a commit to address this, including a test case that fails without the fix

**Review by Rigidity — APPROVED** [review]

**Pull request merged**